pkg/ottl/parser.go

```diff
@@ -11,6 +11,9 @@ import (
 	"github.com/alecthomas/participle/v2"
 	"go.opentelemetry.io/collector/component"
 	"go.uber.org/zap"
+
+	// Embed the timezone database so this code works across platforms.
+	_ "time/tzdata"
```

Comment:
this will add roughly 400kB to binary size

Reply:
I'm not sure why this is required to be added to the parser.  The location logic is handled via `time.LoadLocation(*location)` inside `timeutils.GetLocation`.  Is it not enough to pass the location string to that function?

Reply:
I actually just ran into an issue with this function this morning. [`timeutils.GetLocation`](https://pkg.go.dev/time#LoadLocation) reads from the system's local timezone database. I'm not sure whether this is an ideal solution (especially considering this database is apparently updated somewhat frequently), but I would guess it is probably the most portable.

Reply:
Should the import live in timeutils instead of ottl?

Reply:
i think best practice is to import it in main package not in a library, so i could move this. Thing evan describes is exactly the case why we're importing this. It should work ok on linux, maybe on macos, but windows would be problematic.

Reply:
I think @djaglowski's solution is best: https://github.com/open-telemetry/opentelemetry-collector/issues/9991

Reply:
>should work ok on linux, maybe on macos, but windows would be problematic for sure.

If this is an issue for windows only it can also be done as a build flag in the release pipeline. Windows builds could include it by adding the `-tags=timetzdata` flag.

Reply:
not only, if you don't have data locally it will be an issue on any other system as well

Reply:
i removed import here and based on discussion mentioned above, we will include it in a builder probably. 
also added a not about importing tzdata when building from source.  
delegating rest of the how to include tzdata information to `contrib/builder`